src/js/components/Anchor.js

```diff
@@ -35,7 +50,7 @@ const Anchor = props => {
       href={props.href}
       target={props.target}
       onClick={props.onClick}>
-      {icon}
+      {(props.primary || props.icon)? icon : null}
```

Comment:
I don't think we need the test here. "icon" will already be undefined if we don't want it.